src/spesh/arg_guard.c

```diff
@@ -309,7 +309,7 @@ static void add_nodes_for_callsite(MVMThreadContext *tc, MVMSpeshArgGuard *tree,
         ? add_nodes_for_typed_argument(tc, tree, candidates, cc.cs, cc.typed_idxs,
                 MVM_VECTOR_ELEMS(cc.typed_idxs), next_type_index(cc.cs, -1), 0,
                 end_node)
-        : end_node;
+        : (MVMuint32)end_node;
```

Comment:
Same here, I'd think making end_node an MVMuint32 in the first place should work out fine. `add_result_node` already returns an MVMuint32

Reply:
Yeah, those are all better. Force-pushed an update. That's what I get for making late night PRs.